CMakeLists.txt

```
@@ -396,6 +397,14 @@ if(FORCE_ASSERT)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DDUCKDB_FORCE_ASSERT")
 endif()
 
+if(MOLD_LINKER)
+  if(OS_NAME STREQUAL "linux")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fuse-ld=mold")
+  else()
+    message(FATAL_ERROR "MOLD Linker is only support on Linux")
+  endif()
+endif()
+
```

```suggestion
if(MOLD_LINKER)
  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fuse-ld=mold")
endif()
```

I would prefer to remove the error, since people might have licence for mold for MacOS / available platform might change in the future. And in any case the compiler I believe would fail, so failure will not be silent.